pkg/mediorum/server/server_basic_auth.go

```diff
@@ -73,7 +73,7 @@ func (ss *MediorumServer) requireUserSignature(next echo.HandlerFunc) echo.Handl
 			})
 		} else {
 			// check it is for this upload id
-			if sig.Data.UploadID != id {
+			if id != "" && sig.Data.UploadID != id {
```

Comment:
What's the use case where the ID is empty? Or does that throw a different error?

Reply:
ID is empty when creating a new upload.
ID exists when editing an existing upload.

Before client would send `X-User-Wallet-Addr` header on create and we'd blindly use that.
But now we're using this signed envelope for both create and update.